### DISCUSSION TITLE
How to get GenPollText widget update

### DISCUSSION THREAD

**Discussion by K9H9** [qtile/qtile#3207] [Q&A] [answered]
I want to have the output of python script in the bar, and it should be updated every 5 to 10 seconds.
I made shell script that just runs the python script and managed to get the output to the bar when python script is only outputting one line. Problem is that the python script updates/prints new line every second, so it does not show up in the bar
Here's the GenPollText widget:
```
widget.GenPollText(update_interval=5,func=lambda: subprocess.check_output("/home/My_username/Code/SpotifyQtile/autostart.sh").decode("utf-8"),foreground="#ffffff")
```
When i run only shell script, it works normally
```
#!/bin/bash

python /home/My_username/Code/SpotifyQtile/main.py
exit 0

```

**Comment by m-col** [comment, marked as answer]
Do you often change the code in the script? If not, you could import it into your config and pass the imported function directly to `GenPollText` as the `func`.

**Reply by K9H9** [reply]
No i don't, it just gets current song from spotify.
Didn't though that at all, works perfectly now.

**Reply by m-col** [reply]
Great!
